tmail-backend/integration-tests/jmap/jmap-integration-tests-common/src/main/scala/com/linagora/tmail/james/common/FolderFilteringActionSetMethodContract.scala

```diff
@@ -0,0 +1,525 @@
+      .isEqualTo(
+        s"""[
+           |  "FolderFilteringAction/get",
+           |  {
+           |    "notFound": [],
+           |    "list": [{
+           |      "id": "$taskId",
+           |      "status": "completed",
+           |      "processedMessageCount": 1,
+           |      "successfulActions": 0,
+           |      "failedActions": 0,
+           |      "maximumAppliedActionReached": false
+           |    }]
+           |  },
+           |  "#0"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def createShouldFailOnTeamMailbox(server: GuiceJamesServer): Unit = {
+    val teamMailbox = TeamMailbox(DOMAIN, TeamMailboxName("marketing"))
+    server.getProbe(classOf[TeamMailboxProbe])
+      .create(teamMailbox)
+      .addMember(teamMailbox, BOB)
+    val teamMailboxId: MailboxId = server.getProbe(classOf[MailboxProbeImpl])
+      .getMailboxId(TEAM_MAILBOX_NAMESPACE, Username.fromLocalPartWithDomain("team-mailbox", DOMAIN).asString(), "marketing")
+    val message: Message = Message.Builder
+      .of
+      .setSubject("matchedRules")
+      .setBody("testmail", StandardCharsets.UTF_8)
+      .build
+    server.getProbe(classOf[MailboxProbeImpl])
+      .appendMessage(BOB.asString, teamMailbox.mailboxPath, AppendCommand.from(message))
+
+    // Assume user set rules via Filter/set
+    server.getProbe(classOf[JmapGuiceCustomProbe])
+      .setRulesForUser(BOB,
+        Rule.builder
+          .id(Rule.Id.of("1"))
+          .name("My first rule")
+          .conditionGroup(Rule.Condition.of(Rule.Condition.FixedField.SUBJECT, Rule.Condition.Comparator.CONTAINS, "matchedRules"))
+          .action(new Rule.Action.Builder().setMoveTo(Optional.of(new Rule.Action.MoveTo("Processed"))))
+          .build)
+
+    val response: String = `given`
+      .body(
+        s"""{
+           |  "using": ["urn:ietf:params:jmap:core", "com:linagora:params:jmap:filter", "urn:apache:james:params:jmap:mail:shares"],
+           |  "methodCalls": [
+           |    ["FolderFilteringAction/set",
+           |      {
+           |        "create": {
+           |          "c1": {"mailboxId": "${teamMailboxId.serialize()}"}
+           |        }
+           |      },
+           |      "c1"
+           |    ]
+           |  ]
+           |}""".stripMargin)
+    .when
+      .post
+    .`then`
+      .statusCode(SC_OK)
+      .contentType(JSON)
+      .extract
+      .body
+      .asString
+
+    assertThatJson(response)
+      .inPath("methodResponses[0]")
+      .isEqualTo(
+        s"""[
+           |  "FolderFilteringAction/set",
+           |  {
+           |    "notCreated": {
+           |      "c1": {
+           |        "type": "notFound",
+           |        "description": "${teamMailboxId.serialize()} can not be found"
+           |      }
+           |    }
+           |  },
+           |  "c1"
+           |]""".stripMargin)
+  }
+
+  @Test
+  def createShouldFailWhenTargetOtherUserMailbox(server: GuiceJamesServer): Unit = {
```

Comment:
Please add a test showing tht it succeeds on other user when using sharing capabilities